### PR TITLE
feat(spur-cli): filter squeue by node with -w/--nodelist

### PR DESCRIPTION
**Pull request by shiv-tyagi**
## Summary

Adds `squeue -w/--nodelist` to show only jobs allocated on the given nodes, matching Slurm.

- The value is a hostlist expression: plain comma list (`node1,node2`), compacted ranges (`node[001-003]`), mixed (`node[1,3,5-7]`), and multiple bracket terms (`gpu[01-04],cpu[01-02]`). It is expanded CLI-side via `spur_core::hostlist::expand`, so an invalid or empty value fails with a clear error before any RPC.
- Expanded node names travel over a new appended `GetJobsRequest.nodes` field (proto tag 7, backward compatible). The controller keeps a job only when its `allocated_nodes` intersect the filter, so pending jobs (no allocation) are excluded, as in Slurm.

## Design notes

- Collapsed the `ClusterManager::get_jobs` filter parameters into a `JobFilter` struct. Adding the node dimension would have pushed the positional signature past the `too_many_arguments` lint; the struct also reads clearly at call sites (`JobFilter { nodes, ..Default::default() }`). All existing callers (server, REST, scheduler loop, accounting, tests) were migrated.
- Known limitation (left intentionally): combining `-w` with `-j <array_parent_id>` matches against the synthesized parent's `allocated_nodes`, which reflect only the lowest task. This mirrors how the aggregate already presents its nodelist and is out of scope here.

## Testing

- Unit tests: hostlist expansion parity (comma list, ranges, mixed, invalid/empty) in `squeue.rs`; `get_jobs` node-filter behavior (intersection, empty-filter no-op, pending excluded, union, unknown node) in `cluster.rs`.
- `cargo clippy --workspace --exclude spur-ffi --all-targets --locked`: clean.
- `cargo test --locked`: full suite passes.
- Live bare-metal cluster (4 nodes): pinned jobs to node1/node2/node3, verified `-w` for single node, empty node, compacted ranges, comma lists, full range, and clean nonzero-exit errors for invalid/empty input.

### CODE REVIEW

File: crates/spurctld/src/cluster.rs [hunk 1]
```diff
@@ -988,35 +1004,35 @@ impl ClusterManager {
...
-            if !states.is_empty() && !states.contains(&j.state) {
+            if !filter.states.is_empty() && !filter.states.contains(&j.state) {
+                return false;
+            }
+            if !filter.nodes.is_empty()
+                && !j
+                    .allocated_nodes
+                    .iter()
+                    .any(|n| filter.nodes.iter().any(|f| f == n))
+            {
```
[the remark this answers is not in the document]
Comment (shiv-tyagi) [reply crates/spurctld/src/cluster.rs]: Done in daedc029: build a HashSet from the node filter once per get_jobs call, so the per-job check is O(allocated_nodes).

**Review by biluriuday — APPROVED** [review]
LGTM

The JobFilter refactor is the right call. Collapsing six positional args (that wall of None, None, None, None) into a documented struct at cluster.rs:423-437 kills a real footgun and makes call sites self-documenting (JobFilter { states: &[...], ..Default::default() }). All ~15 call sites migrated cleanly.
Fail-fast client-side validation. expand_node_filter (squeue.rs:384-399) expands and validates before any RPC, consistent with the existing -t/-S parsing pattern. Bad hostlists get a clear local error, not a confusing connect failure.
Correct Slurm semantics. Intersection on allocated_nodes (cluster.rs:1010-1017) naturally excludes pending jobs, matching squeue -w. Verified applied to both the normal scan and the synthesized array-parent path.
Backward-compatible proto. nodes appended as tag 7 (proto/slurm.proto:479-480) — safe additive repeated string, no persisted-state/WAL impact (get_jobs/JobFilter aren't serialized).
Good, real tests. get_jobs_filters_by_node (cluster.rs) covers single/union/unknown/pending-excluded; CLI tests cover padding preservation, multi-term, and empty/invalid rejection.
Actionable items (senior-architect pass)
